File: strings/7 Group Anagrams.py
# Group Anagrams

# 1
def groupAnagrams(words):
  hash_map = {}

  for word in words:
    sorted_word = "".join(sorted(word))
    if sorted_word in hash_map:
      hash_map[sorted_word].append(word)
    else:
      hash_map[sorted_word] = [word] 

  return list(hash_map.values())

File: strings/7 Group Anagrams.js
// Group Anagrams

// #1
function groupAnagrams(words) {
  const hashMap = {}
  const res = []

  for (let word of words) {
    const sortedWord = [...word].sort().join('')

    if (hashMap.hasOwnProperty(sortedWord)) {
      hashMap[sortedWord].push(word)
    } else {
      hashMap[sortedWord] = [word]
    }
  }

  for (prop in hashMap) {
    res.push(hashMap[prop])
  }

  return res
 }

//  #2
function groupAnagrams(words) {
  const hashMap = {}

  for (let word of words) {
    const sortedWord = [...word].sort().join('')

    if (sortedWord in hashMap) {
      hashMap[sortedWord].push(word)
    } else {
      hashMap[sortedWord] = [word]
    }
  }

  return Object.values(hashMap)
 }